### PR TITLE
[MRG+1] Modified gpr so that kernel uses noise

### PR DESCRIPTION
**Pull request by SoManyUsernamesTaken**
See https://github.com/scikit-optimize/scikit-optimize/issues/559

### CODE REVIEW

File: skopt/learning/gaussian_process/gpr.py [hunk 1]
```diff
@@ -186,7 +186,7 @@ def fit(self, X, y):
         if self.kernel is None:
             self.kernel = ConstantKernel(1.0, constant_value_bounds="fixed") \
                           * RBF(1.0, length_scale_bounds="fixed")
-        elif self.noise == "gaussian":
+        if self.noise == "gaussian":
             self.kernel = self.kernel + WhiteKernel()
```
Comment (betatim) [inline skopt/learning/gaussian_process/gpr.py]: is there a way to check if `self.kernel` already contains a `WhiteKernel`? Or does someone know what happens when you add two `WhiteKernel`s together?

**Comment by betatim** [comment]
Thanks for this PR. Could you also add a small test so that this bug doesn't come back?

File: skopt/learning/gaussian_process/gpr.py (hunk 1 above)
Comment (SoManyUsernamesTaken) [reply skopt/learning/gaussian_process/gpr.py]: Not easily anyway. However not much will happen - it will just make optimisation a bit slower as there is one extra useless hyperparameter to perform gradient descent over, but you will still get the same minimum

**Comment by SoManyUsernamesTaken** [comment]
Added test based on example code from the issue

**Comment by betatim** [comment]
Looks good to me! Will hit merge in a few hours, just giving others a chance to look/comment.

**Comment by SoManyUsernamesTaken** [comment]
By the way, should it be default behaviour to assume Gaussian noise? GPR's should be able to discover the amount of noise automatically in most situations

**Comment by betatim** [comment]
I don't remember why that choice was made. It would make sense to allow noise by default given we focus on noisy black box functions. Maybe @MechCoder or @glouppe remember? We should do that in a new PR though.

**Comment by betatim** [comment]
Merged! Thanks for spotting this and fixing it.
